packages/api/src/tasks/checkAlerts/index.ts

```diff
@@ -101,11 +101,16 @@ const fireChannelEvent = async ({
     throw new Error('Team not found');
   }
 
-  if ((alert.silenced?.until?.getTime() ?? 0) > Date.now()) {
+  // Re-fetch the alert to get the latest silenced state. Other alert properties
+  // (threshold, interval, groupBy, etc.) are stable during execution and do not
+  // need to be refetched. We only check the silence state here to ensure we respect
+  // user silencing actions that may have occurred after the alert task was queued.
+  const freshAlert = await Alert.findById(alert.id);
```

Comment:
As per [this issue](https://linear.app/clickhouse/issue/HDX-2281/alert-further-decouple-mongo-from-the-alert-check-loop-oss-repo), we have removed all direct mongo/docdb references from the alerting code and are using the alert provider to fetch data from mongo.

It's possible that original motivation for that ticket ("We're going to eventually have to avoid having alert job pods connecting back to mongo/documentdb.") is no longer true, in which case this is fine. @dhable do you know if this is still a requirement?